### ISSUE TITLE
Always run the native image generator on the module path 

### ISSUE THREAD

**Issue by christianwimmer** [oracle/graal#4924]
### TL;DR

Always run the native image on the module path, i.e., the image builder is a proper module and all its internals are shielded by the module system. Remove the legacy compatibility mode that can currently be enabled via the environment variable `USE_NATIVE_IMAGE_JAVA_PLATFORM_MODULE_SYSTEM=false`

### Goals

Reduce the internal complexity of the image generator by removing the old mode where the image generator itself is on the class path and not a module. This mode cannot be used when the application itself is modularized.

### Non-Goals

* This change does not require that the application itself is on the module path. It will still be possible to build images for non-modularized applications that are on the class path.
* No impact on reflection and introspection of modules at image run time: while full support for those is an eventual goal, it is not tracked under this issue.

### Details

The image generator runs on the module path by default already for most images that are built. A few notable exceptions exist, for example images that contain Truffle languages currently require the image generator to be on the class path. After all of these exception cases are fixed, the old mode of running the image generator on the class path will be removed.
